internal/pkg/component/github/github.go

```diff
@@ -118,20 +153,7 @@ func (c *Component) GetBranch() (string, error) {
 	return branch, nil
 }
 
-func (c *Component) GetToken() (string, error) {
-	ghAppInstallationTokenCacheMutex.Lock()
-	defer ghAppInstallationTokenCacheMutex.Unlock()
-
-	if ghAppInstallationTokenCache == nil || ghAppInstallationTokenCacheID != c.Timestamp {
-		ghAppInstallationTokenCache = NewCache()
-		ghAppInstallationTokenCacheID = c.Timestamp
-	}
-
-	appInstallations, err := c.getAppInstallations()
-	if err != nil {
-		return "", fmt.Errorf("failed to get GitHub App installations: %w", err)
-	}
-
+func (c *Component) getMyInstallationID(appInstallations []AppInstallation) (int64, error) {
```

Comment:
```suggestion
func (c *Component) getInstallationID() (int64, error) {
```
No necessary to pass the installations, you can call `c.getAppInstallations()` inside the function directly.